src/Microsoft.AspNetCore.Mvc.Core/Formatters/TextOutputFormatter.cs

```diff
@@ -165,6 +165,17 @@ public sealed override Task WriteResponseBodyAsync(OutputFormatterWriteContext c
         /// <returns>A task which can write the response body.</returns>
         public abstract Task WriteResponseBodyAsync(OutputFormatterWriteContext context, Encoding selectedEncoding);
 
+        internal static IList<StringWithQualityHeaderValue> GetAcceptCharsetHeaderValues(OutputFormatterWriteContext context)
+        {
+            var request = context.HttpContext.Request;
+            if (StringWithQualityHeaderValue.TryParseList(request.Headers[HeaderNames.AcceptCharset], out IList<StringWithQualityHeaderValue> result))
+            {
+                return result;
+            }
+
+            return null;
```

Comment:
nitpick: consider using `Array.Empty<string>()` here as a sentinel. It's basically free and it's a little friendlier to work with empty collections rather than null collections.

It looks like the surrounding code already tolerates a null value

Reply:
I would prefer to not do that, as that'll result in object creation even in a situation where there is no header we're interested in. Plus, as you've noticed, the current implementation tolerates a null value.

Reply:
Array.Empty will not create an object. It stores the instance on a cache, internally. That's what @rynowak meant with it's basically free.

Reply:
I see. Thanks!

Reply:
Done!